sdk/core/core-http/lib/policies/proxyPolicy.ts

```diff
@@ -26,6 +26,14 @@ function loadEnvironmentProxyValue(): string | undefined {
     return process.env[Constants.HTTP_PROXY];
   } else if (process.env[Constants.HTTP_PROXY.toLowerCase()]) {
     return process.env[Constants.HTTP_PROXY.toLowerCase()];
+  } else if (process.env[Constants.NO_PROXY]) {
+    return process.env[Constants.NO_PROXY];
+  } else if (process.env[Constants.NO_PROXY.toLowerCase()]) {
+    return process.env[Constants.NO_PROXY.toLowerCase()];
+  } else if (process.env[Constants.ALL_PROXY]) {
+    return process.env[Constants.ALL_PROXY];
```

Comment:
.NET preserves both HTTP and HTTPS proxies separately and falls back to `ALL_PROXY` if one (or both) is not set: https://github.com/Azure/azure-sdk-for-net/blob/73551953eb31c81f66c44970a46c7f0d1005067a/sdk/core/Azure.Core/src/Pipeline/Internal/HttpEnvironmentProxy.cs

Should we be doing something smart like that too? i.e. Storing both proxies and picking the one that matches the protocol being used?

.NET also has some tests, which we should probable have too: https://github.com/Azure/azure-sdk-for-net/blob/d3d19ff0057bb3443cf39929b8e79d3ba208bd85/sdk/core/Azure.Core/tests/HttpEnvironmentProxyTest.cs